queries.sql

```diff
@@ -100,4 +100,80 @@ WHERE owners.full_name = 'Dean Winchester' AND escape_attempts = 0;
 
 --- Most pets owner
 SELECT owners.full_name, COUNT(animals.id) AS count FROM animals
-JOIN owners ON owners.id = animals.owner_id GROUP BY owners.full_name ORDER BY count DESC LIMIT 1;
\ No newline at end of file
+JOIN owners ON owners.id = animals.owner_id GROUP BY owners.full_name ORDER BY count DESC LIMIT 1;
+
+--- Who was the last animal seen by William Tatcher?
+
+SELECT animals.name AS pet_name, vets.name AS vet_name, visits.date_of_visit
+FROM animals
+JOIN visits ON visits.animal_id = animals.id
+JOIN vets ON vets.id = visits.vet_id
+WHERE vets.name = 'William Tatcher' 
+ORDER BY visits.date_of_visit DESC LIMIT 1;
+
+---How many different animals did Stephanie Mendez see?
+
+SELECT COUNT(DISTINCT animals.name) AS checked_count
+FROM animals 
+JOIN visits ON visits.animal_id = animals.id 
+JOIN vets ON vets.id = visits.vet_id 
+WHERE vets.name = 'Stephanie Mendez';
+
+---List all vets and their specialties, including vets with no specialties.
+
+SELECT vets.name AS vet , species.name AS species FROM vets
+LEFT JOIN specializations ON vets.id = specializations.vet_id
+LEFT JOIN species ON species.id = specializations.species_id;
+
+---List all animals that visited Stephanie Mendez between April 1st and August 30th, 2020.
+
+SELECT animals.name AS pet_name, visits.date_of_visit FROM animals 
+JOIN visits  ON animals.id = visits.animal_id
+JOIN vets ON vets.id = visits.vet_id
+WHERE vets.name = 'Stephanie Mendez'
+AND visits.date_of_visit BETWEEN '2020-04-01' AND '2020-08-30';
+
+---What animal has the most visits to vets?
+
+SELECT animals.name AS pet_name, COUNT(visits.date_of_visit) AS visits_count
+FROM animals JOIN visits ON visits.animal_id = animals.id 
+GROUP BY animals.name 
+ORDER BY visits_count 
+DESC LIMIT 1;
+
+---Who was Maisy Smith's first visit?
+
+SELECT animals.name AS pet_name, visits.date_of_visit
+FROM animals 
+JOIN visits ON visits.animal_id = animals.id
+JOIN vets ON vets.id = visits.vet_id
+WHERE vets.name = 'Maisy Smith'
+ORDER BY visits.date_of_visit ASC LIMIT 1;
+
+---Details for most recent visit: animal information, vet information, and date of visit.
+
+SELECT animals.name As pet_name, animals.weight_kg, animals.date_of_birth,
+animals.neutered,vets.name AS vet, vets.date_of_graduation,
+visits.date_of_visit FROM animals 
+JOIN visits ON visits.animal_id = animals.id
+JOIN vets ON vets.id = visits.vet_id
+ORDER BY visits.date_of_visit DESC LIMIT 1;
+
+---How many visits were with a vet that did not specialize in that animal's species?
+
+SELECT COUNT(visits.animal_id) AS unspecialized_visits_count FROM visits
+JOIN vets ON visits.vet_id = vets.id
+JOIN animals ON visits.animal_id = animals.id
+JOIN specializations ON specializations.species_id = vets.id
+WHERE specializations.species_id != animals.species_id;
```

Comment:
- [ ] this query returns a number of 8 visits according to your screenshots however, looking at the data in the tables maisy smith has had 9 visits.

Reply:
According to the data I inserted it's the value is 8
![Visits](https://user-images.githubusercontent.com/30289772/236648057-c4550f55-b5bc-409c-8c49-6422f0d10f4f.PNG)